### ISSUE TITLE
🐛 BUG: `process.env.NODE_ENV` in codeblocks breaks Markdown parsing

### ISSUE THREAD

**Issue by naiyerasif** [withastro/astro#2280]
### What version of `astro` are you using?

0.22.1

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

The following codeblock which uses `process.env.NODE_ENV` breaks the markdown parsing.

    ```
    export const environment = {
      env: process.env.NODE_ENV
    }
    ```

with the following error.

```sh
12:20:47 PM [vite] Error when evaluating SSR module /home/projects/github-jh8snw/src/pages/index.astro:
Error: Parse failure: Unexpected token (1:97)
Contents of line 1: export const metadata = { "headers": [], "source": "\n```\nexport const environment = {\n  env: "development"\n}\n```\n", "html": '<pre class="language-null"><code data-astro-raw class="language-null">export const environment = {\n  env: "development"\n}</code></pre>' };
    at ssrTransform (/home/projects/github-jh8snw/node_modules/astro/vendor/vite/dist/node/chunks/dep-35df7f96.js:55943:15)
    at doTransform (/home/projects/github-jh8snw/node_modules/astro/vendor/vite/dist/node/chunks/dep-35df7f96.js:56785:48) (x2)
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-jh8snw?devtoolsheight=33&file=src/pages/posts/index.md

**Comment by jonathantneal** [comment]
Verified. It only happens when the value matches `process.env.NODE_ENV` with variable-like boundaries. This means `process.env.NODE_ENVS` or `process.env.NODE_EN` will not throw, and neither will `sprocess.env.NODE_ENV.A`. However, `process.env.NODE_ENV.A` will throw.

**Comment by FredKSchott** [comment]
Confirmed fixed in Astro v0.24.0: https://stackblitz.com/edit/github-zqhkh9?file=src%2Fpages%2Findex.astro&on=stackblitz
